src/components/OrganizationAutocomplete/OrganizationAutocomplete.tsx

```diff
@@ -0,0 +1,87 @@
+import React, { useState } from 'react';
+import { useDebounce } from 'react-use';
+import { Except } from 'type-fest';
+import { AutocompleteField, AutocompleteFieldProps } from '../form';
+import { useOrganizationAutocompleteQuery } from './OrganizationAutocomplete.generated';
+
+type OrganizationAutocompleteProps<
+  T,
+  Multiple extends boolean | undefined,
+  DisableClearable extends boolean | undefined,
+  FreeSolo extends boolean | undefined
+> = Except<
+  AutocompleteFieldProps<T, Multiple, DisableClearable, FreeSolo>,
+  'options' | 'loading'
+>;
+
+export function OrganizationAutocomplete<
+  T,
+  Multiple extends boolean | undefined,
+  DisableClearable extends boolean | undefined,
+  FreeSolo extends boolean | undefined
+>(
+  props: OrganizationAutocompleteProps<T, Multiple, DisableClearable, FreeSolo>
+) {
+  const [value, setValue] = useState('');
+  const [options, setOptions] = useState<any[]>([]);
+  const [selectedOptions, setSelectedOptions] = useState<any[]>([]);
+  const [searching, setSearching] = useState(false);
+
+  // if we don't want to show any initial options, it'd be nice not to have to call the api before input
+  const { refetch } = useOrganizationAutocompleteQuery({
+    variables: {
+      input: {
+        filter: {
+          name: '',
+        },
+      },
+    },
+  });
+
+  useDebounce(
+    async () => {
+      if (value === '') return;
+      const { data } = await refetch({
+        input: {
+          filter: {
+            name: value,
+          },
+        },
+      });
+      setOptions(
+        data?.organizations?.items.map((item) => item.name.value) || []
+      );
```

Comment:
We want the options to be the objects, that way we can get the ID from it.

Reply:
What are we using the id for? And I think the options passed to Autocomplete have to be strings.

Reply:
The IDs are what are sent to the API. Nope they don't have to be strings. Several props are functions given a `T` generic object. In this case `T` will be `Organization`.

Reply:
By options do you mean the literal `options` prop?